rfcs/20180918-functions-not-sessions-20.md

```diff
@@ -0,0 +1,1002 @@
+
+NOTE: In TensorFlow 1.x, eager execution is enabled by <code>[tf.enable_eager_execution()](https://www.tensorflow.org/api_docs/python/tf/enable_eager_execution)</code>. Once invoked, all public API endpoints that consume or produce symbolic <code>Tensor</code> objects begin to produce and consume <code>Tensor</code> objects that are backed by a concrete value. See the "Research and Experimentation" section at [www.tensorflow.org/tutorials](http://www.tensorflow.org/tutorials) for an introduction.
+
+
+### A few things of note
+
+
+
+*   This change **only** applies to the TensorFlow **Python** frontend
+    *   [TensorFlow.js](https://js.tensorflow.org/) is already "eager by default".
+    *   [Switf4TensorFlow](https://github.com/tensorflow/swift) has [similar design goals](https://github.com/tensorflow/swift/blob/master/docs/DesignOverview.md#swift-for-tensorflow-design-overview), doing away with the define-then-run style of TensorFlow graphs.
+    *   Most other language bindings ([Java](https://www.tensorflow.org/api_docs/java/reference/org/tensorflow/package-summary), [C++](https://www.tensorflow.org/api_guides/cc/guide), [Go](https://godoc.org/github.com/tensorflow/tensorflow/tensorflow/go), others) are mostly targeting deployment of defined models in applications. While an imperative style might help simplify model development and training in these languages, doing so is explicitly out of scope for TensorFlow 2.0. The notion of graphs and sessions will remain in them, as well as in the stable [C API](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/c/c_api.h). In these APIs, the lifetime of program state (like variables) will continue to be tied to the lifetime of the `Session`.
+*   Users of **<code>Estimator</code>** will see no change
+    *   Canned Estimators are black boxes that create and train models. Enabling eager execution will have no effect on their usage. This is true today.
+    *   The <code>model_fn</code> of a regular (non-canned) <code>Estimator</code> will remain as a graph construction function.
+*   <strong>[SavedModel](https://www.tensorflow.org/guide/saved_model#save_and_restore_models)</strong> will continue to be the format encouraged for exporting trained models
+    *   Crudely speaking, a SavedModel encapsulates a Graph, a checkpoint of variable values, and some metadata like signature information (names of input and output tensors).
+    *   A path will be provided to easily export models in this format (e.g., via <code>tf.keras.Model.save()</code>). There may be instances where converting the Python code to a graph is not trivial, in which case, exporting to a SavedModel (and thus a Graph) will fail.
+
```

Comment:
Could you give an example of such a case where exporting to a `SavedModel` will fail?

Reply:
Added a note. Thanks.